scripts/hud/boardrig.lua

```diff
@@ -328,14 +330,131 @@ end
+                -- cell wasn't already added to our list
+                if distance <= sprintNoise and cell
+                    and (boardrig:canPlayerSee( x, y )
+                    or unit:getPlayerOwner()._ghost_cells[ simquery.toCellID( x, y ) ])
+                    and not array.find(cells, cell) then
+                    table.insert( cells, cell )
+                end
+            end
+        end
+    end
+
+    -- get only the tiles with hearing-enabled, known enemy units on them; this can include ghosts
+    for i, cell in pairs(cells) do
+        if sim:canPlayerSee(sim:getPC(), cell.x, cell.y) then
+            -- check real units
+            for i, unit in pairs(cell.units) do
+                if unit:getTraits().hasHearing and sim:canPlayerSeeUnit(sim:getPC(), unit)
+                and unit:getPlayerOwner() ~= sim:getPC() then
+                    table.insert(units, unit)
+                end
+            end
+        else
+            -- check ghost units
+            local ghost_cell = sim:getPC()._ghost_cells[ simquery.toCellID( cell.x, cell.y ) ]
+            for i, unit in pairs(ghost_cell.units) do
+                if unit:getTraits().hasHearing and unit:getPlayerOwner() ~= sim:getPC() then
+                    table.insert(units, unit)
+                end
+            end
+        end
+    end
+
+    unit.UITR_props = {}
+
+    for i, target in pairs(units) do
+        local targetrig = boardrig:getUnitRig(target:getID())
+        local x, y = target:getLocation() 
+        wx, wy = boardrig:cellToWorld(x, y) 
+        -- tested: highlighting the unit itself (would have to be undone via rig:refresh())
+        --[[targetrig._prop:setRenderFilter({
+            shader = KLEIAnim.SHADER_FOW,
+            r = 142/255,
+            g = 247/255, 
+            b = 142/255,
+            a = 1,
+            lum = 1.3
+        })]]
+        local prop = targetrig:createHUDProp("kanim_soundbug_overlay_alarm", "character", "alarm_loop", boardrig:getLayer("ceiling"), nil, wx, wy )
+
+        prop:setSymbolModulate("cicrcle_wave", 247/255, 247/255, 142/255, 1 )
+        prop:setSymbolModulate("line_1", 247/255, 247/255, 142/255, 1 )
+        prop:setSymbolModulate("ring", 247/255, 247/255, 142/255, 1 )
+        prop:setSymbolModulate("attention_ring", 247/255, 247/255, 142/255, 1 )
+    
+        table.insert(unit.UITR_props, prop)   
+    end
+
+    -- highlight each tile in the list
+    --local hiliteID = boardrig:hiliteCells( preCells, SPRINT_HILITE_CLR )
+
+    --boardrig._UITRSprintHilite = hiliteID
+end
+
+-- ===
+
 local oldChainCells = boardrig.chainCells
 local oldUnchainCells = boardrig.unchainCells
 
 function boardrig:chainCells(cells, color, ...)
-    local id = oldChainCells(self, cells, color, ...)
     local selectedUnit = self._game.hud:getSelectedUnit()
     local sim = self:getSim()
 
+    -- sprint noise preview logic here; there's an unhilite in unchainCells as well -Sizzle
```

Comment:
Did the "unhilite in `unchainCells`" get dropped? I'm seeing issues where the prop is never cleared if you Tab to another agent or otherwise cancel the movement targeter.
I'd recommend a pattern where the list of props is stored inside of `self._chainCells[id]` using the `id` return value from `oldChainCells`, because the game knows when to clean up that cluster of props by calling `unchainCells` with the same `id`.
  * `.props` and `.duration` declared by vanilla.
  * `.fgProps` is a new field I added for the overwatch warnings)